data/service/api/v1/datasets_data_create.go

```diff
@@ -95,10 +94,18 @@ func DataSetsDataCreate(dataServiceContext dataService.Context) {
 		return
 	}
 
+	var provenance *data.Provenance
+	if claims, err := dataServiceContext.AuthClient().ServerSessionTokenClaims(ctx); err != nil {
```

Comment:
This doesn't look correct - we should get the client of the access token that was passed in the `x-tidepool-session-token` header.

Reply:
Yeah, you caught this for review before it was ready, sorry... But all the same, you're right, that's the wrong token from which to grab the client id. 😄 

I'll move it to a draft.